lib/util/determine-files.js

```diff
@@ -0,0 +1,81 @@
+let path = require("path");
+let fs = require("fs");
+
+/* Returns a function that determines files to process
+ *
+ * This is useful for pipelines that process a directory of files.
+ * It takes a source directory and optionally a predicate that is used to filter
+ * the determined files. It automatically filters invisible files.
+ *
+ * If the returned function is called without any arguments, it will return a
+ * Promise with an array of all files that matched.
+ * If it is called with an array of paths, it will return a Promise with an
+ * array of all of those files that matched.
```

Comment:
As you might have guessed from the comments above, I believe this documentation could be simplified. In that sense, it's probably a good thing that my brain is very tired at the moment (otherwise I might not have noticed it quite as much). However, it also means I can't come up with a holistic proposal right now - will do that some other time.